src/Platforms/AbstractPlatform.php

```diff
@@ -3882,6 +3896,28 @@ public function getTimeTypeDeclarationSQL(array $column)
         throw Exception::notSupported(__METHOD__);
     }
 
+    /**
+     * Adds a parameter to a TIME-type SQL declaration to indicate precision
+     *
+     * @param string  $decl   The SQL declaration to update e.g. TIME, DATETIME
+     * @param mixed[] $column The column definition
+     */
+    protected function addVariablePrecisionTimeSQL(string $decl, array $column): string
+    {
+        if (
+            ! isset($column['length']) ||
+            ! is_numeric($column['length']) ||
+            (int) $column['length'] < 0 ||
+            ((int) $column['length'] === 0 && $this->ignoreZeroTimePrecision())
+        ) {
+            return $decl;
+        }
+
+        $precision = min($this->getTimeMaxPrecision(), (int) $column['length']);
```

Comment:
Why does the DBAL need to silently reduce the precision specified by the user?

Reply:
Should we throw instead? That would mean, if we use a precision of 9 in a schema, we could deploy this to schema to Oracle but not to e.g. MariaDB. But maybe that's better than a silent downgrade.

Then again, how would I tell DBAL to pick the maximum precision for the current database?

Reply:
Why would you want DBAL to define your schema?